provider/baked/src/zerotrie.rs

```diff
@@ -55,17 +55,29 @@ pub struct Data<M: DataMarker> {
 }
 
 impl<M: DataMarker> super::DataStore<M> for Data<M> {
-    fn get(&self, id: DataIdentifierBorrowed) -> Option<&'static <M>::Yokeable> {
+    fn get(
+        &self,
+        id: DataIdentifierBorrowed,
+        attributes_prefix_match: bool,
+    ) -> Option<&'static <M>::Yokeable> {
         use writeable::Writeable;
         let mut cursor = self.trie.cursor();
         let _is_ascii = id.locale.write_to(&mut cursor);
         if !id.marker_attributes.is_empty() {
             cursor.step(ID_SEPARATOR);
             id.marker_attributes.write_to(&mut cursor).ok()?;
+            loop {
+                if let Some(v) = cursor.take_value() {
```

Comment:
nit: please document this. it's unclear to me how `take_value()` will fail once but succeed the next time?

Reply:
`.probe(0)` mutates the cursor.

Reply:
tbh, I don't find this API very intuitive. `probe` feels similar to `peek`, which is idempotent, maybe `step_nth` would be a better name.

Reply:
I like `.step_nth` a lot

Here's the PR where I first added it. I asked for other names but didn't get any at the time :)

https://github.com/unicode-org/icu4x/pull/4725